packages/app/obojobo-repository/server/services/permissions.js

```diff
@@ -22,6 +22,30 @@ const fetchAllUsersWithPermissionToDraft = async draftId => {
 	return users.map(u => new UserModel(u))
 }
 
+const userHasPermissionToCopy = async (userId, draftId) => {
+	const results = await Promise.all([
+		draftIsPublic(draftId),
+		userHasPermissionToDraft(userId, draftId)
+	])
+
+	return results[0] || results[1]
+}
+
+const draftIsPublic = async draftId => {
+	const publicLibCollectionId = '00000000-0000-0000-0000-000000000000'
+
```

Comment:
Nice, I think this will be useful.  We probably need to make sure the value of `publicLibCollectionId` is in a single location in the repository and used as a const everywhere.